vividus-plugin-csv/src/main/java/org/vividus/csv/transformer/CsvTableTransformer.java

```diff
@@ -18,62 +18,119 @@
 import org.vividus.util.ExamplesTableProcessor;
 
 public class CsvTableTransformer implements ExtendedTableTransformer
 {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CsvTableTransformer.class);
+    private static final String VARIABLE_NAME_PROPERTY = "variableName";
+    private static final String PATH_PROPERTY = "path";
+    private static final String DEPRECATED_CSV_PATH_PROPERTY = "csvPath";
+
     private final CSVFormat defaultCsvFormat;
+    private final VariableContext variableContext;
 
-    public CsvTableTransformer(CSVFormat csvFormat)
+    public CsvTableTransformer(CSVFormat csvFormat, VariableContext variableContext)
     {
         this.defaultCsvFormat = csvFormat;
+        this.variableContext = variableContext;
     }
 
     @Override
-    public String transform(String tableAsString, TableParsers tableParsers, TableProperties properties)
+    public String transform(String tableAsString, TableParsers tableParsers, TableProperties tableProperties)
     {
         checkTableEmptiness(tableAsString);
-        String csvPath = properties.getMandatoryNonBlankProperty("csvPath", String.class);
+        Properties properties = tableProperties.getProperties();
+
+        Map.Entry<String, String> entry = processCompetingMandatoryProperties(properties,
+                VARIABLE_NAME_PROPERTY, PATH_PROPERTY, DEPRECATED_CSV_PATH_PROPERTY);
+        String sourceKey = entry.getKey();
+        String sourceValue = entry.getValue();
+        checkDeprecatedProperty(sourceKey);
 
         CSVFormat csvFormat = defaultCsvFormat;
-        String delimiter = properties.getProperties().getProperty("delimiterChar");
+        String delimiter = properties.getProperty("delimiterChar");
         if (delimiter != null)
         {
             int delimiterLength = delimiter.length();
             isTrue(delimiterLength == 1, "CSV delimiter must be a single char, but value '%s' has length of %d",
                     delimiter, delimiterLength);
             csvFormat = csvFormat.builder().setDelimiter(delimiter.charAt(0)).build();
         }
+
+        List<Map<String, String>> result = VARIABLE_NAME_PROPERTY.equals(sourceKey)
+                ? readCsvFromVariable(csvFormat, sourceValue)
+                : readCsvFromFile(csvFormat, sourceValue);
+        return ExamplesTableProcessor.buildExamplesTable(result.get(0).keySet(), extractValues(result),
+                tableProperties, true);
+    }
+
+    private List<Map<String, String>> readCsvFromFile(CSVFormat csvFormat, String path)
+    {
         try
         {
-            List<CSVRecord> result = new CsvReader(csvFormat).readCsvFile(findResource(getClass(), csvPath));
-            return ExamplesTableProcessor.buildExamplesTable(result.get(0).toMap().keySet(), extractValues(result),
-                    properties, true);
+            return new CsvReader(csvFormat).readCsvString(loadResourceOrFileAsString(path));
```

Comment:
it's better to introduce `CsvReader#readCsv(bytes[])` method to avoid conversions to string and back